rockPaperScissors.py

```diff
@@ -3,48 +3,51 @@
 
 import random
 
-def gameLogic(player, computer, playerScore, computerScore):
-    playerIndex = RPS.index(player)
-    computerIndex = RPS.index(computer)
-
-    gameIncrement = 0
-    if (playerIndex != computerIndex):
-        gameIncrement = 1
-        playerNextIndex = (playerIndex + 1) % len(RPS)
-        
-        if (playerNextIndex == computerIndex):
-            computerScore += 1
-        else:
-            playerScore += 1
+def gameLogic(player1Choice, player2Choice):
+    if player1Choice == player2Choice:
+        return 0
 
-    return gameIncrement, playerScore, computerScore
+    playerNextIndex = (player1Choice + 1) % 3
+    
+    winner = 2 if playerNextIndex == player2Choice else 1
+    
+    return winner
 
-maxGames=3
-RPS=['rock','paper','scissors']     
 
-computerScore=0   
-playerScore=0  
-gameCount=0   
+RPS = ['rock', 'paper', 'scissors']
+maxGames = 3
+computerScore = 0
+playerScore = 0
+gameCount = 0
+
 print('I challenge you to rock paper scissors, best 2 out of 3!')
-while gameCount < maxGames:    #loops for 3 rounds
-    print('rock paper scissors!')
-    player=input()
-    computer=random.choice(RPS)
-    print(computer)
-    gameIncrement, newComputerScore, newPlayerScore = gameLogic(computer, player, computerScore, playerScore)
 
-    if gameIncrement == 0:
+while gameCount < maxGames:    #loops for 3 rounds
+    playerChoice = input('rock paper scissors! Make your choice:')
+    try:
```

Comment:
Dunno if you've gotten to error handling yet. Basically, lines 27 - 31 try to find the index of the player's choice. if it exists, great. the game moves on. If the user types in something that isn't in that list, calling ```RPS.index(<value that is not in the list>)``` will throw an error, and the program will move to the ```except``` block, where I've told it to mock the user and re-start from the beginning of the while loop

Reply:
I haven't gotten an error message yet, but I know it will crash with input that isn't in the list. I was going to get around to it at some point in the future as an exercise in debugging and getting used to the git work flow thing

Reply:
Try/except (or try/catch in most other languages) is the classic way of handling this sort of scenario. There are others though. If you want to make your head hurt, check out result functors.